### PR TITLE
Update SDK for multichain and support WAGMI

### PR DESCRIPTION
**Pull request by iam-robi**
SDK modifications to support multichain.

**IMPORTANT NOTICE** 
This is the first stage of all PR that will implement multichain support. The **npm package needs to be updated** with this code with your choice of version. we suggest to move to a new major version 2.0.0-beta for example. 

Once the  sdk is published we will be able to PR for the following repos: 

- components => _will need a npm publish as well_  for the interface to work
- interface
- tokens / tokenlists

Please notice we recommend `beta` on package name because contracts for wagmi will be used for testing and republish to put the right multisigs , initial farms after tests are validated.

### CODE REVIEW

File: src/entities/currency.ts [hunk 1]
```diff
@@ -33,5 +40,6 @@ export class Currency {
   }
 }
 
-const CAVAX = Currency.CAVAX
+//$ const CAVAX = Currency.CAVAX
+const CAVAX = Currency.CURRENCY
```
Comment (SarjuHansaliya) [inline src/entities/currency.ts:44]: I think this change will break interface / components. so we need to release this sdk version as major version and also update interface/components

File: src/entities/fractions/currencyAmount.ts [hunk 2]
```diff
@@ -17,9 +16,10 @@ export class CurrencyAmount extends Fraction {
   /**
    * Helper that calls the constructor with the ETHER currency
    * @param amount ether amount in wei
+   * @param chainId
    */
-  public static ether(amount: BigintIsh): CurrencyAmount {
-    return new CurrencyAmount(CAVAX, amount)
+  public static ether(amount: BigintIsh, chainId: ChainId): CurrencyAmount {
```
Comment (SarjuHansaliya) [inline src/entities/fractions/currencyAmount.ts:21]: what if we init with default chainId like `chainId: ChainId = ChainId.AVALANCHE`, so it doesn't break existing code.

File: src/router.ts [hunk 3]
```diff
@@ -73,16 +74,16 @@ export abstract class Router {
    * @param trade to produce call parameters for
    * @param options options for the call parameters
    */
-  public static swapCallParameters(trade: Trade, options: TradeOptions | TradeOptionsDeadline): SwapParameters {
-    const etherIn = trade.inputAmount.currency === CAVAX
-    const etherOut = trade.outputAmount.currency === CAVAX
+  public static swapCallParameters(trade: Trade, options: TradeOptions | TradeOptionsDeadline, chainId: ChainId): SwapParameters {
```
Comment (SarjuHansaliya) [inline src/router.ts:77]: same here as my last comment

File: src/entities/currency.ts (hunk 1 above)
Comment (iam-robi) [reply src/entities/currency.ts:44]: yes these changes are breaking and we have the interface/ components code that fits this version ready to PR once the SDK is published. this is probably better anyway to release a major version. 
this interface branch works with the 2.0 sdk https://github.com/wagmiog/interface/tree/implement-multichain
and we have a live interface here: https://quirky-kowalevski-0e801a.netlify.app/#/beta/airdrop

**Comment by iam-robi** [comment]
@SarjuHansaliya we have made the change on your comments to have chainId: ChainId = ChainId.AVALANCHE as default chain. 

for CAVAX we plan to make a new PR after everything is integrated as it's not working w/ a simple replace. also it could be done later in sync with contracts repo for complete harmonization.

**Comment by SarjuHansaliya** [comment]
awesome thanks @iam-robi

**Review by SarjuHansaliya — APPROVED** [review]

**Review by HariSeldon23 — APPROVED** [review]

**Comment by SarjuHansaliya** [comment]
@iam-robi / @jb1011 i think this PR has conflict, can you please resolve ?

**Pull request merged**